ssz/src/main/java/org/ethereum/beacon/ssz/access/SSZListAccessor.java

```diff
@@ -1,44 +1,70 @@
 package org.ethereum.beacon.ssz.access;
 
 import org.ethereum.beacon.ssz.access.SSZCompositeAccessor.CompositeInstanceAccessor;
+import org.ethereum.beacon.ssz.type.SSZType;
+import tech.pegasys.artemis.util.bytes.BytesValue;
 
 /**
- * Handles collections of homogeneous objects (e.g. java array, {@link java.util.List}, etc)
- * which includes both ssz types: vectors and lists
- * List accessor is responsible of accessing list elements, their type and new instance creation
+ * Handles collections of homogeneous objects (e.g. java array, {@link java.util.List}, etc) which
+ * includes both ssz types: vectors and lists List accessor is responsible of accessing list
+ * elements, their type and new instance creation
  *
- * This interface also serves as its own {@link CompositeInstanceAccessor}
- * since there is normally no information to cache about the list type
- * (as opposed to {@link SSZContainerAccessor})
+ * <p>This interface also serves as its own {@link CompositeInstanceAccessor} since there is
+ * normally no information to cache about the list type (as opposed to {@link SSZContainerAccessor})
  */
 public interface SSZListAccessor extends SSZCompositeAccessor, CompositeInstanceAccessor {
 
-  interface ListInstanceBuilder extends CompositeInstanceBuilder {
+  @Override
+  int getChildrenCount(Object value);
 
-    /**
-     * Appends a new child to the building list
-     */
-    void addChild(Object childValue);
+  /**
+   * Returns atomic size of a list. Normally it's 1:1 to element size, see {@link
+   * #getChildrenCount(Object)}, but assuming structures like Bit List we have elements that are
+   * smaller than byte (atoms) which are accessible as bytes
+   */
+  default int getAtomicChildrenCount(Object value) {
```

Comment:
This change breaks an invariant of `getChildrenCount` that should return a number of bits for bit list and vector like it does for other collections. I think we need a calculator for size in bytes for each type which defaults to `sizeof(element) * getChildrenCount(value)`. And, also, we might need a stream that returns encoded bytes for collection.

Reply:
I would suggest to change the `SSZType.getSize()` to return rational value (or `double` for simplicity, but you need to be careful with converting to int) and add a `SSZBitType` as the `Bitlist` element type with the size of `1/8`

Reply:
An extra complexity that this calculation should be different for serialization and hasher